edgelet/iotedge/src/support_bundle.rs

```diff
@@ -75,45 +85,78 @@ where
 {
     pub fn new(
         log_options: LogOptions,
-        location: OsString,
         include_ms_only: bool,
         verbose: bool,
         iothub_hostname: Option<String>,
+        output_location: OutputLocation,
         runtime: M,
     ) -> Self {
         SupportBundle {
             runtime,
             log_options,
-            location,
             include_ms_only,
             verbose,
             iothub_hostname,
+            output_location,
         }
     }
 
-    fn make_state(self) -> Result<BundleState<M>, Error> {
-        let file_options =
-            zip::write::FileOptions::default().compression_method(zip::CompressionMethod::Deflated);
-        let location = PathBuf::from(&self.location);
+    fn make_file_state(self) -> Result<BundleState<M, File>, Error> {
+        let file_options = FileOptions::default().compression_method(CompressionMethod::Deflated);
 
-        let zip_writer = zip::ZipWriter::new(
-            File::create(&location)
+        let zip_writer = ZipWriter::new(
+            File::create(Path::new(self.output_location.get_file_location()))
                 .map_err(|err| Error::from(err.context(ErrorKind::SupportBundle)))?,
         );
 
         Ok(BundleState {
             runtime: self.runtime,
             log_options: self.log_options,
-            location,
             include_ms_only: self.include_ms_only,
             verbose: self.verbose,
             iothub_hostname: self.iothub_hostname,
             file_options,
             zip_writer,
+            output_location: self.output_location,
+        })
+    }
+
+    fn make_vector_state(self) -> Result<BundleState<M, Cursor<Vec<u8>>>, Error> {
```

Comment:
Might be cleaner to make this `fn make_state<W>(self, writer: W) -> Result<BundleState<M, W>, Error>` and have the caller pass in the appropriate writer. Then you don't have two methods that are otherwise nearly identical.